core/corehttp/gateway_handler.go

```diff
@@ -125,6 +125,10 @@ func (i *gatewayHandler) getOrHeadHandler(w http.ResponseWriter, r *http.Request
 	i.addUserHeaders(w) // ok, _now_ write user's headers.
 	w.Header().Set("X-IPFS-Path", urlPath)
 
+	//robots-header to prevent search-engines from indexing the gateway because of the duplicate-content problematic when makeing the gateway public
+	w.Header().Set("X-Robots-Tag", "noindex")
```

Comment:
I'm owner of onion.cab as see you  can see here: https://onion.cab/?a=contact
There exist multiple tor2web proxies which are crawleable by google. I can tell you that google assigns every [a-z0-9]{16}.onion domain to exactly one of this tor2web-proxies. The same subdomain at any other of the proxies is indexed but is not realy ranking for anything.
If you have more than one ipfs-gateway crawlable than google would also take the one with the most trust and unrank every other one.
See https://www.google.de/?gws_rd=ssl#q=torbox+(site:*.onion.city+OR+site:*.onion.link+OR+site:*.onion.cab+OR+site:*.onion.to)


The proposed sitemap should in no way include everything from ipfs. It should be fillable by the gateway-owner to show only what belongs to him.